go/cmd/dolt/commands/diff.go

```diff
@@ -909,6 +909,7 @@ func schemaFromCreateTableStmt(createTableStmt string) (schema.Schema, error) {
 			TypeInfo:      typeInfo,
 			Default:       defBuf.String(),
 			Generated:     "",    // TODO
+			OnUpdate:      "",    // TODO
```

Comment:
I don't remember why this is difficult to fix. Why can't we use a  SqlEngine here and issue two "show create table" statements?

Reply:
If it's not trivial, can you do as a followup?

Reply:
I'm not sure what's wrong here (at least with Generated and Virtual). I'm able to create tables with generated and virtual columns and the `SHOW CREATE ...` statement displays them just fine when calling `dolt diff`.

However, there is a bug with `ON UPDATE`:
We can parse ...`ON UPDATE CURRENT_TIMESTAMP()`, but not `ON UPDATE (CURRENT_TIMESTAMP())`

Should be a simple vitess parser fix, will be fixed in a follow.
Tracking issue here: https://github.com/dolthub/dolt/issues/7129#issuecomment-1854798812